articles/embedded.md

```diff
@@ -0,0 +1,318 @@
+---
+layout: default
+title: Embedded ROS 2.0
+abstract:
+  This article captures requirements, design ideas and related works on a tiny ROS 2.0 stack for microcontrollers.
+published: true
+author: 'In alphabetic order: [Adam Dąbrowski](https://github.com/adamdbrw), [Borja Outerelo](https://github.com/BorjaOuterelo), [Ingo Lütkebohle](https://github.com/iluetkeb), [Ralph Lange](https://github.com/ralph-lange), [Víctor Mayoral Vilches](https://github.com/vmayoral), ... and many more from Embedded ROS 2 Interest Group. Please feel free to add your name.'
+---
+
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+While ROS 2 can run on Linux-class embedded devices, smaller 
+microcontrollers (MCU) are still a challenge. For example, ARM Cortex-M3 
+and -M4 class MCUs are popular for motion control and sensor integration, 
+but have little RAM and storage. ARM Cortex-M3 is widely used in many 
+sensors due to its optimized power consumption capabilities and ultra low 
+power modes. These microcontrollers are often running with small batteries, 
+e.g. for environmental sensors or small consumer robots, which requires
+efficient power management. The embedded ROS 2 effort addresses these devices.
+
+This article documents feature wishes, requirements, design ideas and 
+related work on this undertaking.
+
+## Important Differences / Assumptions
+
+One of the major difference for the embedded stack is to be able to
+run not (only) on Linux, but also on Real-Time Operating Systems, or
+even "bare metal" (no OS).
+
+## Wishlist
+
+A number of things that would be great to have, lets call it a wishlist. Feel free to add stuff ;-)
+
+### W-SEAMLESS: *Seamless integration with ROS&nbsp;2*
+
+Support all the communication and configuration mechanisms expected,
+e.g. topics, services, parameters (including dynamic modification),
+the life-cycle, etc.
+
+It should not be noticeable to the rest of the system that part of
+it runs on a micro-controller.
+
+### W-PORTABILITY: *Portability of ROS 2-based software*
+
+Use the same API as on ROS 2.
+
+Note that the feasibility of this wish has been questioned.
+
+### W-DEVICES *Support of a broad spectrum of device classes*
+
+Support a broad range of MCU device classes, starting from a 
+few tens kilobyte of RAM, up to megabytes.
+
+### W-RTOS: *Support of a broad spectrum of RTOS(s)* 
+
+The stack shall be adaptable to different RTOS (e.g., NuttX, FreeRTOS, 
+Zephyr) and provide abstractions with regard to the application 
+software to facilitate porting application components to between RTOSs.
+
+### W-COMM: *Support of prevalent communication technologies* 
+
+To cover the broad range of use-cases for MCUs in robotics, the embedded 
+ROS 2 stack shall be usable with the default ROS 2 middleware standard 
+DDS, simple (custom) serial communication protocols just as common 
```

Comment:
Imo the statement "usable with" is not specific enough in terms of goals. In the meaning of "compatible with on the wire" there is simple no other choice than DDS. In the meaning of "compatible using e.g. a bridge" I don't see much value in mentioning it - that part is explicitly covered by `W-SEAMLESS`. Technically even ROS 1 is compatible in that sense using the `ros1_bridge`.

Reply:
What's meant here is "usable" in the same way that you can exchange rmw_-implementations now.

Not sure how to rephrase that so it becomes clearer. Maybe just mentioning rmw would do the trick? Other suggestions welcome.